dex/pair/src/safe_price.rs

```diff
@@ -47,10 +51,10 @@ impl<M: ManagedTypeApi> NestedDecode for PriceObservation<M> {
         let weight_accumulated = u64::dep_decode(input)?;
         let recording_round = u64::dep_decode(input)?;
 
-        let lp_supply_accumulated = if !input.is_depleted() {
-            BigUint::dep_decode(input)?
+        let (recording_timestamp, lp_supply_accumulated) = if !input.is_depleted() {
+            (u64::dep_decode(input)?, BigUint::dep_decode(input)?)
```

Comment:
This will NOT work if you upgrade an existing pair with safe price. If you want to add new fields to a struct, always add them at the end, as they're encoded/decoded in that order.